pkg/security/module/module.go

```diff
@@ -143,7 +176,9 @@ func (m *Module) EventDiscarderFound(rs *rules.RuleSet, event eval.Event, field
 
 // HandleEvent is called by the probe when an event arrives from the kernel
 func (m *Module) HandleEvent(event *sprobe.Event) {
```

Comment:
As discussed on Slack, your idea of using an array of rulesets and swapping the ruleset might help to avoid a mutex. Keeping the two rulesets shouldn't be a big deal, but if it is, we can always delete the one that is not in use when the first event evaluated by the new ruleset goes through.